src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -28,66 +27,61 @@ public class FindCommandParser implements Parser<FindCommand> {
      */
     public FindCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_COMPANY_NAME, PREFIX_STATUS, PREFIX_TAG);
+                ArgumentTokenizer.tokenize(args, PREFIX_COMPANY_NAME, PREFIX_ROLE, PREFIX_STATUS, PREFIX_TAG);
 
         if (!argMultimap.getPreamble().isEmpty()) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
 
         if (!argMultimap.getValue(PREFIX_COMPANY_NAME).isPresent()
+                && !argMultimap.getValue(PREFIX_ROLE).isPresent()
                 && !argMultimap.getValue(PREFIX_STATUS).isPresent()
                 && !argMultimap.getValue(PREFIX_TAG).isPresent()) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
 
+        List<String> names = new ArrayList<>();
+        List<String> roles = new ArrayList<>();
+        List<String> statuses = new ArrayList<>();
+
         List<String> nameKeywords;
+        List<String> roleKeywords;
         List<String> statusKeywords;
         List<String> tagKeywords;
 
-        if (argMultimap.getValue(PREFIX_COMPANY_NAME).isPresent()) {
-            CompanyName companyName = ParserUtil.parseCompanyName(argMultimap.getValue(PREFIX_COMPANY_NAME).get());
-            nameKeywords = split(companyName.fullCompanyName);
-        } else {
-            nameKeywords = Collections.emptyList();
+        List<String> nameList = argMultimap.getAllValues(PREFIX_COMPANY_NAME);
+        for (String name : nameList) {
+            names.add(ParserUtil.parseCompanyName(name).fullCompanyName);
+        }
+        nameKeywords = this.split(names);
+
+        List<String> roleList = argMultimap.getAllValues(PREFIX_ROLE);
+        for (String role : roleList) {
+            roles.add(ParserUtil.parseRole(role).fullRole);
         }
+        roleKeywords = this.split(roles);
 
-        if (argMultimap.getValue(PREFIX_STATUS).isPresent()) {
-            Status status = ParserUtil.parseStatus(argMultimap.getValue(PREFIX_STATUS).get());
-            statusKeywords = split(status.fullStatus);
-        } else {
-            statusKeywords = Collections.emptyList();
+        List<String> statusList = argMultimap.getAllValues(PREFIX_STATUS);
+        for (String status : statusList) {
+            statuses.add(ParserUtil.parseStatus(status).fullStatus);
         }
+        statusKeywords = this.split(statuses);
```

Comment:
Just to confirm, does this code mean that the syntax of finding by multiple filters `find n/google facebook` or `find n/google n/facebook`?

Reply:
Because the name field can take in multiple words, either way works.

Reply:
How the parser works is it will first run getAllValues which returns a List<String> where every element is a string containing all the words in every name field. I then loop through the string and split every string by the whitespaces, essentially like a flatMap.

Reply:
We can deal with the inconsistencies in v1.3. There is a need to standardise across other commands like `add` and `edit` as well. Currently, for v1.2b, `find` looks good.